### ISSUE TITLE
Support for specifying a custom URL that has a searx instance

### ISSUE THREAD

**Issue by DonkeeeyKong** [fastaddons/SearchResultPreviews#22]
Hello. First of all: Thank you for this great add-on! I have been using it for years!

**Is your feature request related to a problem? Please describe.**
Right now the only searx instance that is supported is searx.org. But there is a [whole list of public instances](https://searx.space/) plus many more non public installations including people self-hosting there own private instance.

**Describe the solution you'd like**
It would be great if it was possible to add a custom URL with a searx instance in the settings.

**Describe alternatives you've considered**
The option to alter the searx.org entry to another instance.

Thanks for considering! :)

**Comment by Juraj-Masiar** [comment]
I'll see what I can do!
Just to brainstorm here, what I need to do is:
1. allow users to add custom domain
2. assign that domain to an existing domain (or maybe allow extending existing domain with a custom URL which will inherit the "rules" and "selectors" needed for finding the search results)
3. add a code that will ask "Request access to `<new host>`" permission. 
Hmmm, this one sounds like a show-stopper. From what I can see in the [docs](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/permissions/request), I need to define all hosts manually upfront:
>Permissions must come from the set of permissions defined in the [optional_permissions](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/optional_permissions) manifest.json key.

So addon cannot request permanent access to a new host that user enters, it must always be pre-specified in the manifest file in the addon.

But I could just add those domains from the link you've posted, similarly as I'm already doing it for 100 google domains :)

**Comment by DonkeeeyKong** [comment]
Hi, thanks a lot for the quick answer!

> But I could just add those domains from the link you've posted, similarly as I'm already doing it for 100 google domains :)

This is a good improvement already, it would still exclude all non-public or selfhosted instances though.

> So addon cannot request permanent access to a new host that user enters, it must always be pre-specified in the manifest file in the addon.

I found this answer on Stack Overflow, where somebody was asking for help with the same problem: https://stackoverflow.com/a/71913707

Is including `<all_urls>` in the `optional_permissions` of manifest.json a problem? The specific permissions per website would still be requested only if the user specifically asks for that and not automatically.

**Comment by Juraj-Masiar** [comment]
Yes, using "all_urls" should help, but there is big downside for Firefox users:
![image](https://github.com/user-attachments/assets/b56d59f9-379c-4350-9f23-bea84f7dbf80)

The blue/green "attention required" dot will be displayed on every single page (currently it's displayed only on pages where the search is actually supported).

More info in these super long threads:
https://bugzilla.mozilla.org/show_bug.cgi?id=1851083
https://bugzilla.mozilla.org/show_bug.cgi?id=1766026

Chrome currently doesn't show the attention icon (I think only Firefox and Safari does).
Maybe we should request this on bugzilla, to not show the dot for the "all_urls" since requesting attention for every page is not a good design.

**Comment by Juraj-Masiar** [comment]
Actually... there seem to have been some development and the attention icon is no longer enabled for Firefox 130 and above?
https://bugzilla.mozilla.org/show_bug.cgi?id=1905931
(I'm developing with Firefox ESR 128, so I'll still see it)

I'll investigate :)

**Comment by DonkeeeyKong** [comment]
> Actually... there seem to have been some development and the attention icon is no longer enabled for Firefox 130 and above? https://bugzilla.mozilla.org/show_bug.cgi?id=1905931 (I'm developing with Firefox ESR 128, so I'll still see it)

I just tried and visited a website that's supported by the plugin but not enabled by me and there is no such attention dot on Firefox 132.0.1. Only when I click on the plugin's icon I see a banner to enable the plugin on this website.

**Comment by DonkeeeyKong** [comment]
> Actually... there seem to have been some development and the attention icon is no longer enabled for Firefox 130 and above? https://bugzilla.mozilla.org/show_bug.cgi?id=1905931 (I'm developing with Firefox ESR 128, so I'll still see it)
> 
> I'll investigate :)

Any news on this? :)

**Comment by Juraj-Masiar** [comment]
Sadly, no.
Each time I start working on a new update, it's due to some hotfix and in the limited time I have a always manage to add only some small fixes/upgrades and this issue requires much more work. Plus the fact that it would negatively affect existing ESR users makes me really want to postpone it until ESR 140 is released and existing users are upgraded - which is usually at the end of September, so not that far away!

At the same time, I'm really not happy about adding `all_urls` permission, even if as optional one, it's still visible during installation and on the store page as *"This add-on may also ask to: Access your data for all websites"*.
I would rather add all the domains listed in the `searx.space`. This would be also much easier to implement since I could reuse all of the existing code.
